magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/SqlHelper.scala

```diff
@@ -252,4 +260,13 @@ object SqlHelper {
       case OpaValueNumber(bigDec)   => AspectQueryBigDecimal(bigDec)
     }
   }
+
+  /**
+    * safely escape an identifier string (e.g. column, table names)
+    * sqls"" (SQLSyntax object) will always treats external input values as binding parameters. Thus, not suitable for identifier
+    * @param name identifier string
+    * @return escaped identifier as SQLSyntax object
+    */
+  def escapeIdentifier(name: String): SQLSyntax =
+    SQLSyntax.createUnsafely('"' + name.replaceAll("\"", "\"\"") + '"')
```

Comment:
I think there are two risks we might want to look at:
- 1: SQL injection risk (this function is for prevent this)
- 2: we shouldn't allow user to select `any` column they want
The function is used for processing query reference like aspect0, aspect1 etc. 
without this function we have to put like:
SQLSyntax.createUnsafely(ref) like [here](https://github.com/magda-io/magda/blob/ad93d911def22470f9bf1552089877793170b741/magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordPersistence.scala#L1441)

So `escapeIdentifier` here  just add some extra protection (for protecting SQL injection risk) otherwise we have to call `SQLSyntax.createUnsafely` which is not sql injection safe.

We only have two kind of references  is passed to `escapeIdentifier `:
- ColumnNamePrefixType.PREFIX_TEMP --- before is passed to `SQLSyntax.createUnsafely` directly
- aspectSelectors --- generated by [aspectIdsToSelectClauses](https://github.com/magda-io/magda/blob/ad93d911def22470f9bf1552089877793170b741/magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordPersistence.scala#L1711). Its logic seems guarantee only token like `aspect0`, `aspect1` will be produced.

What's your thoughts on this?

Reply:
> SQL injection risk (this function is for prevent this)

This is what I'm most worried about - because although we're replacing characters here and that _should_ prevent SQL Injection, we're replacing a very very hard to bypass method of mitigating it (parameters) with something (perhaps imho) much easier to bypass... e.g. what if postgres adds another escape character, or we missed something, or the implementation of `replaceAll` changes.

If we're only passing in PREFIX_TEMP and aspectSelectors then that's good, but it's still really easy for someone to come along later and pass user input to it. Can we create a special type for those and make it so that this function only accepts that type? And maybe make it so that rather than making identifiers with escape characters safe, it just throws an exception (because I don't think we really need to support escape characters do we?).

Reply:
Yeah~ I don't think we have used any special chars in identifiers.
How about we remove this `escapeIdentifier ` function and revert it back to the `SQLSyntax.createUnsafely`?

Reply:
Yeah if you like :)